### PR TITLE
Revive sandbox + New API changes

### PR DESCRIPTION
**Pull request by shamansir**
`Tween` / `Modifier` API improved a lot, now they're proper classes instead of function wrappers, and `Tween` is actually just extends `Modifier`. Also, nice names for tweens and easings instead of weird constants.
Sandbox now uses new API, including these changes.
First examples were updated to a new API, next examples will be updated later.

Will add it to `gh-pages` branch few hours later.

### CODE REVIEW

**Comment by shamansir** [comment]
@mcm69 please review

**Comment by shamansir** [comment]
I will merge it to show it in action, it is still a subject to review, will do a separate PR for fixes.

File: src/anm/animation/tween.js [hunk 1]
```diff
@@ -27,152 +34,284 @@ var errors = require('../errors.js');
...
-               };
-    mod.data = data_block_fn; // FIXME
+var Tween = Modifier;
+
+function createTween(type, value) {
+    var mod = new Modifier(null, C.MOD_TWEEN); // mod.func will be set to `null` first
+    var def = Registry[type];
+    mod.def = def;
+    mod.func = function(t, dt, duration) { if (mod.$tween) mod.$tween.call(this, t, dt, duration); };
+    mod.is_tween = true;
```
Comment (mcm69) [inline src/anm/animation/tween.js]: what's this for, isn't it already created as `MOD_TWEEN`?

File: src/anm/animation/tween.js [hunk 2]
```diff
@@ -27,152 +34,284 @@ var errors = require('../errors.js');
...
+Tween.register = function(type, definition) {
+    definition = is.fun(definition) ? { func: definition } : definition;
+    definition.from = definition.from || Tween.DEFAULT_FROM;
+    definition.to   = definition.to   || Tween.DEFAULT_TO;
+    Registry[type] = definition;
+    Tween[type] = function(value) { return createTween(type, value); };
+}
 
-var Tweens = {};
+Tween._$ = createTween;
```
Comment (mcm69) [inline src/anm/animation/tween.js]: what's this?

File: src/anm/animation/tween.js (hunk 1 above)
Comment (shamansir) [inline src/anm/animation/tween.js]: true, will remove it. thanks!

File: src/anm/animation/tween.js (hunk 2 above)
Comment (shamansir) [inline src/anm/animation/tween.js]: Probably it's not needed to be exposed. It was planned to act similar to `Element._$`, which allows user to do something like `var element = Element._$` and then do nice chaining like: `element(<name>).band().fill()...`. So user decides what name should the chain root have.

**Pull request merged**